sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/JsonSupport.java

```diff
@@ -158,14 +193,29 @@ public static <T> T decodeJson(Class<T> valueClass, Any any) {
     }
   }
 
+  private static <T> T migrate(Class<T> valueClass, ByteString decodedBytes, int fromVersion, JacksonMigration jacksonMigration) throws IOException {
+    JsonNode jsonNode = objectMapper.readTree(decodedBytes.toByteArray());
+    JsonNode newJsonNode = jacksonMigration.transform(fromVersion, jsonNode);
+    return objectMapper.treeToValue(newJsonNode, valueClass);
+  }
+
+  private static int parseVersion(String typeUrl) {
+    if (typeUrl.contains("#")) { //TODO can we assume that there will be ony one "#" ??
```

Comment:
I suppose we can, right?

Reply:
Safer to do a regex match for something like `.*#(\d+)$`?

Reply:
Yeah, regexp is an option, but I'm always worried that it might be more costly. Maybe I shouldn't care too much in this case...

Reply:
`lastIndexOf('#')` should be efficient

Reply:
updated